### DISCUSSION TITLE
mobx-react + injection: wrappedComponent not found

### DISCUSSION THREAD

**Discussion by gcattan** [mobxjs/mobx#3499] [General]
Hello, 
I have just finished the mobx 4 to mobx 6 migration, and I noticed a difference in behavior.
In the image below we are rendering a component called `BeSelectorField` and we keep a reference to this component inside a member called `fieldHasPopup`.

![image](https://user-images.githubusercontent.com/6229031/184433896-78ac4c45-934b-40ce-97f9-778304496c1d.png)

The `BeSelectorField` is a class component with the annotation `@inject`.

In Mobx 4, the reference `fieldHasPopup` was an `Injector`, and to access the component itself we had to do `fieldHasPopup.wrappedComponent`.

Since we migrated to Mobx 6, `fieldHasPopup` is directly an instance of `BeSelectorField`.

Changing the syntax is not a issue, but I just want to check with you that this difference is expected. Thanks!

**Comment by urugator** [comment]
Yes, we now forward the ref automatically:
https://github.com/mobxjs/mobx/blob/f0e066f427d573cbef4b92f3310eb069c3aca205/packages/mobx-react/src/inject.ts#L19-L30
I assume `React.forwardRef` did not exist or it just wasn't possible for other reason at that time . For completeness:
https://github.com/mobxjs/mobx-react/blob/f8302d81aee6795dd4b042e7408d8f2d38912d03/src/inject.js#L36-L73

EDIT: Not sure what you mean by *"wrappedComponent not found"*, but the component should still be also available as `wrappedComponent` prop:
https://github.com/mobxjs/mobx/blob/f0e066f427d573cbef4b92f3310eb069c3aca205/packages/mobx-react/src/inject.ts#L37

**Reply by gcattan** [reply]
Ok, thanks for the answer! Indeed, in our case, the ref is not of type Injector and there is no `wrappedComponent` property available inside.
So instead of doing `fieldHasProp.wrappedComponent.myAwesomeProperty` (mobx 4) we are doing `fieldHasProp.myAwesomeProperty` directly (since mobx 6).
